### ISSUE TITLE
Classifier.predict() says my pretrained and accurate model is inaccurate.

### ISSUE THREAD

**Issue by animalcroc** [Trusted-AI/adversarial-robustness-toolbox#1627]
**Describe the bug**

I included a code snippet. Basically, I use the example located at <https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/main/examples/get_started_tensorflow.py>, with some variations. Instead of training a model, I create my model and initialize with weights loaded from a pickle file. To verify my model works properly, I call classifier.predict(). Instead of getting a high percentage, i get 8%-10% accuracy, which is not in agreement with the test accuracy I find when I don't use classifier.predict(). For some reason unknown to me, classifier.predict() gives me numbers indicative of random guesses. I assume that my pretrained model is not correctly being read by the aversarial robustness toolbox framework.

**To Reproduce**

`
import tensorflow.compat.v1 as tf
import numpy as np
tf.compat.v1.disable_eager_execution()
#from tensorflow.compat.v1.keras.models import Model
import tensorflow.compat.v1.keras.layers
from tensorflow.compat.v1.keras.models import Sequential
from tensorflow.compat.v1.keras.layers import Dense, Dropout, Activation, Flatten
from art.estimators.classification import TensorFlowClassifier
from art.utils import load_mnist
import pickle

# Step 1: Load the MNIST dataset

(x_train, y_train), (x_test, y_test), min_pixel_value, max_pixel_value = load_mnist()

# load layer weights
with open('C:\\Users\\valentin\\Desktop\\my_name\\file_adv.pkl', 'rb') as f:
    lay_weights = pickle.load(f)
    
    
# Step 2: Create the model
input_ph = tf.placeholder(tf.float32, shape=[None, 28, 28, 1])
labels_ph = tf.placeholder(tf.int32, shape=[None, 10])

model = Sequential()
model.add(Flatten(input_shape=(28, 28, 1)))
model.add(Activation('relu'))
model.add(Dense(300))
model.add(Activation('relu'))
model.add(Dense(200))
model.add(Activation('relu'))
model.add(Dense(150))
model.add(Activation('relu'))
model.add(Dense(150))
model.add(Activation('relu'))
model.add(Dense(100))
model.add(Activation('sigmoid'))
model.add(Dense(10))
model.layers[2].set_weights([lay_weights[0], lay_weights[1]])
model.layers[4].set_weights([lay_weights[2], lay_weights[3]])
model.layers[6].set_weights([lay_weights[4], lay_weights[5]])
model.layers[8].set_weights([lay_weights[6], lay_weights[7]])
model.layers[10].set_weights([lay_weights[8], lay_weights[9]])
model.layers[12].set_weights([lay_weights[10], lay_weights[11]])

#set logits equal to model output
logits=model(input_ph)

#check that my model is accurate
for ind in range(0,10):
    x_sample1=x_test[ind,:,:,:]
    x_sample1=np.expand_dims(x_sample1,0)
    mod_output=model.predict(x_sample1)
    print(np.argmax(mod_output))
    print(np.argmax(y_test[ind]))

sess = tf.Session()
sess.run(tf.global_variables_initializer())

# Step 3: Create the ART classifier

classifier = TensorFlowClassifier(
    clip_values=(min_pixel_value, max_pixel_value),
    input_ph=input_ph,
    output=logits,
    labels_ph=labels_ph,
    learning=None,
    sess=sess,
    preprocessing_defences=[],
)

# Step 5: Evaluate the ART classifier on benign test examples
predictions = classifier.predict(x_test)
accuracy = np.sum(np.argmax(predictions, axis=1) == np.argmax(y_test, axis=1)) / len(y_test)
print("Accuracy on benign test examples: {}%".format(accuracy * 100))

`
**Expected behavior**
I expected to get an accuracy (high 90th percentile) of my model, as reported by classifier.predict(), which agrees with tests results that are obtained when I do not call classifier.tests()

**Screenshots**

**System information (please complete the following information):**
 - Windows 10
 - python 3.7.10 
 - ART version or commit number
 - TensorFlow 2.1 and 2.4.1
